File: implementations/fsharp/tests/Skight.AgentPlatform.FSharp.Tests/WebToolsSpecs.fs
namespace Skight.AgentPlatform.FSharp.Tests

open System
open System.Text.Json
open Expecto
open Skight.AgentPlatform.FSharp

module WebToolsSpecs =

    [<Tests>]
    let tests =
        testList "Web Tools Module Specs" [
            
            testAsync "Feature: Fetch URL Content - Returns clean stripped text" {
                let urlJson = JsonSerializer.Serialize({| url = "https://httpbin.org/html" |})
                let! content = WebTools.fetchUrlContentAsync urlJson |> Async.AwaitTask
                Expect.isFalse (content.Contains("<html")) "Content should not contain raw HTML tags"
                Expect.isFalse (String.IsNullOrWhiteSpace(content)) "Content should not be empty"
            }
        ]

File: implementations/csharp/tests/Skight.AgentPlatform.MSpec.Tests/WebToolsSpecs.cs
using System;
using System.Text.Json;
using Machine.Specifications;
using FluentAssertions;
using Skight.AgentPlatform;

namespace Skight.AgentPlatform.MSpec.Tests
{
    [Subject("Web Tools Module - URL Content Fetching")]
    public class When_agent_fetches_content_from_web_url
    {
        Establish context = () =>
        {
            _urlJson = JsonSerializer.Serialize(new { url = "https://httpbin.org/html" });
        };

        Because of = () =>
        {
            _content = WebTools.FetchUrlContentAsync(_urlJson).GetAwaiter().GetResult();
        };

        It should_return_clean_text_content_without_html_tags = () =>
            _content.Should().NotContain("<html")
                .And.NotBeNullOrWhiteSpace();

        static string _urlJson;
        static string _content;
    }
}
